src/middleware/mod.rs

```diff
@@ -28,11 +29,15 @@ pub struct RequestContext<'a, Data> {
     pub app_data: Data,
     pub req: Request,
     pub params: Option<RouteMatch<'a>>,
-    pub(crate) endpoint: &'a BoxedEndpoint<Data>,
+    pub(crate) endpoint: &'a EndpointData<Data>,
     pub(crate) next_middleware: &'a [Arc<dyn Middleware<Data> + Send + Sync>],
 }
 
 impl<'a, Data: Clone + Send> RequestContext<'a, Data> {
+    pub fn get_config<T: Any + Clone + Send + Sync>(&self) -> Option<&T> {
```

Comment:
Since this is returning a config item, I feel like `get_config_item` would be a more descriptive name.

Reply:
Agreed! I've changed the name.